server/routerlicious/packages/lambdas/src/alfred/interfaces.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IEvent } from "@fluidframework/common-definitions";
+import { IClient, IConnected } from "@fluidframework/protocol-definitions";
+
+/**
+ * Connection details of a client
+ */
+export interface IConnectedClient {
+	/**
+	 *
+	 */
+	connection: IConnected;
+
+	/**
+	 *
+	 */
+	details: IClient;
+
+	/**
+	 * Client protocol versions of standard semver types.
+	 */
+	connectVersions: string[];
```

Comment:
@znewton @hedasilv @pradeepvairamani lastly, for connectVersions, this is only ever used in this function here in aflred/index.ts. I think it's referring to protocol_definitions versions, but not sure:

```
const protocolVersions = ["^0.4.0", "^0.3.0", "^0.2.0", "^0.1.0"];

function selectProtocolVersion(connectVersions: string[]): string | undefined {
	for (const connectVersion of connectVersions) {
		for (const protocolVersion of protocolVersions) {
			if (semver.intersects(protocolVersion, connectVersion)) {
				return protocolVersion;
			}
		}
	}
}
```
https://github.com/ghoshkaj/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/alfred/index.ts#L104-L114

But do you think you could add more details to the docs here?

Reply:
AFAIK this protocol versioning functionality has not been used or updated in the 3+ years I've been working in the Fluid service area. We _should_ be using it, or we will likely want to use it in the future. In the meantime, I don't think there's any additional context we should add. I don't even think the versions align with anything else at all. R11s uses protocol-definitions versions far beyond these accepted values, so it's not related to that.

Some other jsdocs for related properties from protocol-defs:
```ts
{
	/**
	 * Protocol version selected by the server to communicate with the client
	 */
	version: string;

	/**
	 * List of protocol versions supported by the server
	 */
	supportedVersions: string[];
}
{
	/**
	 * Semver list of protocol versions supported by the client ordered in priority of use
	 */
	versions: string[];
}
```

TL;DR: I think the doc you added sums it up well.